### DISCUSSION TITLE
[Bug]: Librechat force me to enter OpenAI key, even  I set OPENROUTER_API_KEY and PALM_KEY in .env and use model on OpenRouter

### DISCUSSION THREAD

**Discussion by ttpro1995** [danny-avila/LibreChat#1292] [Troubleshooting]
### Contact Details

hahattpro@gmail.com

### What happened?

I am not use OpenAI. 
I set OPENROUTER_API_KEY, PALM_KEY in .env

### Steps to Reproduce

copy .env.example to .env 
in file .env
1. Leave OPENAI_API_KEY=user_provided (as it is) 
2. Set openrouter key OPENROUTER_API_KEY=sk-or-v1-my-real-key-i-checked-it-correct
3. Set palm key PALM_KEY=AIzaSyDx9-it-is-correct-for-sure
4. docker compose up
5. http://localhost:3080/c/new 
6. Pick a model that on openrouter. In this case, pick openchat/openchat-7b (which is currently free on openrouter) 
7. The chat bar is block, with text "Set your Key in the Header menu to chat" 

### Screenshots

![image](https://github.com/danny-avila/LibreChat/assets/7196876/1f082be3-8c23-4548-9377-1a32231ebb5d)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
When using OpenRouter, put something in the `OPENAI_API_KEY=` and you won't be asked to enter it. (it can be anything, but do not leave it blank or use `user_provided`)

**Comment by ttpro1995** [comment]
> When using OpenRouter, put something in the `OPENAI_API_KEY=` and you won't be asked to enter it. (it can be anything, but do not leave it blank or use `user_provided`)

What about PALM_KEY, how to switch to palm ?
I pick google/palm-2-chat-bison 
but it end up using Open Router key (which cost $ instead of free with palm key)

**Comment by fuegovic** [comment]
> What about PALM_KEY, how to switch to palm ? I pick google/palm-2-chat-bison but it end up using Open Router key (which cost $ instead of free with palm key)

with PaLM, it's easier to provide the json file in the UI. Look at this guide for the supported method: https://docs.librechat.ai/install/apis_and_tokens.html#googles-palm-2

**Comment by ttpro1995** [comment]
I did provide PALM api key in .env 
Then on the UI, how should i switch to palm mode ? 
It does not show on a list of header/model selection.

**Reply by danny-avila** [reply]
> I did provide PALM api key in .env

There is no API key to set in `.env` for PaLM2. You have to provide the json file

If you've done this already, and you are using docker, you should rebuild: https://docs.librechat.ai/install/docker_compose_install.html#updating-librechat
